checker/check-request-parameters-type-changed.go

```diff
@@ -0,0 +1,86 @@
+package checker
+
+import (
+	"fmt"
+
+	"github.com/tufin/oasdiff/diff"
+)
+
+func RequestParameterTypeChangedCheck(diffReport *diff.Diff, operationsSources *diff.OperationsSourcesMap, config BackwardCompatibilityCheckConfig) []BackwardCompatibilityError {
+	result := make([]BackwardCompatibilityError, 0)
+	if diffReport.PathsDiff == nil {
+		return result
+	}
+	for path, pathItem := range diffReport.PathsDiff.Modified {
+		if pathItem.OperationsDiff == nil {
+			continue
+		}
+		for operation, operationItem := range pathItem.OperationsDiff.Modified {
+			if operationItem.ParametersDiff == nil {
+				continue
+			}
+			for paramLocation, paramDiffs := range operationItem.ParametersDiff.Modified {
+				for paramName, paramDiff := range paramDiffs {
+					if paramDiff.SchemaDiff == nil {
+						continue
+					}
+					typeDiff := paramDiff.SchemaDiff.TypeDiff
+					formatDiff := paramDiff.SchemaDiff.FormatDiff
+					if typeDiff == nil && formatDiff == nil {
+						continue
+					}
+
+					if typeDiff != nil && typeDiff.From == "integer" && typeDiff.To == "number" {
+						continue
+					}
+
+					if typeDiff != nil && typeDiff.To == "string" {
+						// parameters could be change to string anytime
+						continue
+					}
+
+					if formatDiff != nil && (formatDiff.To == nil || formatDiff.To == "") {
+						continue
+					}
+
+					if formatDiff != nil && paramDiff.Revision.Schema.Value.Type == "string" &&
+						(formatDiff.From == "date" && formatDiff.To == "date-time" ||
+							formatDiff.From == "time" && formatDiff.To == "date-time" ||
+							formatDiff.From == "decimal" && formatDiff.To == "uuid") {
```

Comment:
Are you sure this is non-breaking?

Reply:
I am not sure even it is useful and in most implementations are not breaking. But strict checks could fail so I agree with you and removed this posibilites for request parameters and properties changes